supervision/annotators/core.py

```diff
@@ -1346,88 +1438,169 @@ def annotate(
 
         """
         assert isinstance(scene, Image.Image)
+        self._validate_labels(labels, detections)
+
         draw = ImageDraw.Draw(scene)
-        anchors_coordinates = detections.get_anchors_coordinates(
-            anchor=self.text_anchor
-        ).astype(int)
+        labels = self._get_labels_text(detections, labels)
+        text_properties = self._get_text_properties(draw, labels)
+
+        xyxy = self._calculate_label_positions(
+            detections, text_properties, self.text_anchor
+        )
+
```

Comment:
It seems to me that `_get_text_properties` and `_calculate_label_positions` should be merge into a single function that returns numpy array. You can than cut out `xyxy` portion of that numpy array and pass it to `spread_out_boxes`.